### DISCUSSION TITLE
Search functionality with flynt components

### DISCUSSION THREAD

**Discussion by dhananjayharel** [flyntwp/flynt#434] [Q&A] [answered]
I am using flynt theme and its really awesome. 
Our team has developed 50+ components in the site and now we need to add a site search functionality. 
So my question is do we have any third party plugin which can work with flynt theme? 
because contents and residing i components and normal search plugin will not search the contents of the custom components.

**Comment by aaronmeder** [comment, marked as answer]
Hey @dhananjayharel 

We are using [Relevanssi](https://www.relevanssi.com/) for this. The free version is good enough for most of our customer's sites. It's a fast, indexed search.

You must set the "custom fields" option to "visible" in the menu "Settings  > Relevanssi > Index" and then generate an index.

This will then work with the native Wordpress search, so you can still use the Flynt Search component. Access the search page with the `/?s=searchterm` url on your site.

The plugin also allows you to select which post types get index and so on, which can also be helpful.

Greetings from 🇨🇭
Aaron

**Comment by fabianoaljava** [comment]
Hi @dhananjayharel, 

This is probably too late, but I just found this post today and I want to share with the community what I did to include make Flynt's ACF fields searchable.

I've added the following filters in the `functions.php` file.

```
// Modifies the WP search query to include ACF meta performing left join on the posts and postmeta tables
add_filter('posts_join', function ($join) {
    global $wpdb;
    if (is_search()) {
        $join .= ' LEFT JOIN ' . $wpdb->postmeta . ' ON ' . $wpdb->posts . '.ID = ' . $wpdb->postmeta . '.post_id ';
    }
    return $join;
});

// Modifies the WP search query to include ACF meta (metavalue) in the WHERE clause
add_filter('posts_where', function ($where) {
    global $pagenow, $wpdb;
    if (is_search()) {
        $where = preg_replace("/\(\s*" . $wpdb->posts . ".post_title\s+LIKE\s*(\'[^\']+\')\s*\)/", "(" . $wpdb->posts . ".post_title LIKE $1) OR
(" . $wpdb->postmeta . ".meta_value LIKE $1)", $where);
    }
    return $where;
});

// Modifies the WP search query adding a DISTINCT to prevent returning duplicates
add_filter('posts_distinct', function ($where) {
    global $wpdb;
    if (is_search()) {
        return "DISTINCT";
    }
    return $where;
});
```

Not sure if this is the best practice, but it's working for me.

Warm regards,


Fabiano Aljava 🇧🇷🇬🇧
